src/main/java/net/lldp/checksims/token/TokenList.java

```diff
@@ -111,15 +112,20 @@ public String join(boolean onlyValid) {
     }
 
     /**
-     * Peforms a shallow copy of a TokenList, returning an immutable version of the initial list.
+     * Peforms a deep copy of a TokenList, returning an immutable version of the initial list with immutable tokens.
```

Comment:
So, this isn't technically a deep copy, as ImmutableToken is just a wrapper for Token.

To make it a deep copy, you'd clone Tokens before making them immutable